backend/app/api/v1/endpoints/auth.py

```diff
@@ -0,0 +1,343 @@
+        if permission:
+            log.success(f"User with id {id} fetched successfully")
+            return user
+
+    except HTTPException:
+        raise
+
+    except Exception as e:
+        log.error(f"Failed to get user: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to get user: {str(e)}",
+        )
+
+
+@router.put(
+    "/user/{id}",
+    status_code=status.HTTP_200_OK,
+    response_model=UserInResponse,
+)
+async def update_user(
+    id: int,
+    username: str = Form(None),
+    first_name: str = Form(None),
+    last_name: str = Form(None),
+    contact_number: str = Form(None),
+    db: AsyncSession = Depends(get_db),
+    user_id: int = Depends(get_current_user),
+):
+    log.info(f"Attempting to update user with id: {id}")
+    try:
+        user = await fetch_data_by_id(db, User, id)
+        if not user:
+            log.warning(f"User with id {id} does not exist")
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail=f"User with id: {id} does not exist",
+            )
+
+        await check_authorization(user_id, user)
+
+        if username is not None and username != user.username:
+            existing_username = await db.execute(
+                select(User).where(User.username == username).where(User.id != id)
+            )
+            if existing_username.scalar():
+                log.warning(f"Username {username} already exists")
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    detail="Username already exists",
+                )
+            user.username = username
+
+        if first_name is not None:
+            user.first_name = first_name
+
+        if last_name is not None:
+            user.last_name = last_name
+
+        if contact_number is not None:
+            user.contact_number = contact_number
+
+        await update_instance(db, user)
+        log.success(f"User with id {id} updated successfully")
+        return user
+
+    except Exception as e:
+        log.error(f"Failed to update user: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to update user: {str(e)}",
+        )
+
+
+@router.post("/login", status_code=status.HTTP_200_OK)
+async def login(
+    email: str = Form(...),
+    password: str = Form(...),
+    db: AsyncSession = Depends(get_db),
+):
+    log.info(f"Attempting login for email: {email}")
+    try:
+        user = await db.execute(select(User).filter(User.email == email))
+        user = user.scalar_one()
+
+        if not user or not verify_password(password, user.password):
+            log.warning(f"Invalid credentials for email: {email}")
+            raise HTTPException(
+                status_code=status.HTTP_403_FORBIDDEN, detail="Invalid Credentials"
```

Comment:
add a constant Class inside the `core` folder and add these values as variable there _Invalid Credentials_. Prioritize not to use strings in the API layer.

Reply:
Added a file inside the core folder named constants.py which has a class named SystemMessages() containing all the messages from logs to errors. then used them in the API endpoints :

```
except Exception as e:
        log.error(f"{SystemMessages.ERROR_CREATE_USER}: {e}")
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=SystemMessages.ERROR_CREATE_USER_DETAIL,
        )
```